gateway/reverse_proxy.go

```diff
@@ -1705,7 +1789,7 @@ func copyRequest(r *http.Request) *http.Request {
 	}
 
 	if r.Body != nil {
-		r.Body = copyBody(r.Body)
+		r.Body = copyBody(r.Body, false)
```

Comment:
This is the hot path. As server-side (http.Request) bodies don't need closing, a flag is added to trigger the .Body copy into a byte buffer - think of it as a hint for a lazy-alloc-loading.